src/rgw/rgw_lua_request.cc

```diff
@@ -761,6 +761,13 @@ struct RequestMetaTable : public EmptyMetaTable {
       pushstring(L, s->trans_id);
     } else if (strcasecmp(index, "Tags") == 0) {
       create_metatable<StringMapMetaTable<RGWObjTags::tag_map_t>>(L, false, &(s->tagset.get_tags()));
+    } else if (strcasecmp(index, "User") == 0) {
+      const auto id_ptr = &(s->user->get_id());
```

Comment:
`get_id()` just return a reference to an `rgw_user` object, so this pointer to it cannot be null.
this means that you do not need the check in line 766
however, `s->user` is a pointer, so, in theory it may be null.
calling `get_id()` on a null pointer would be a segfailt. so, you probably need to add a check that `s->user` is not null, and do `lua_pushnil()` if it does